python/ray/ml/preprocessor.py

```diff
@@ -72,7 +78,7 @@ def transform(self, dataset: Dataset) -> Dataset:
         Returns:
             ray.data.Dataset: The transformed Dataset.
         """
-        if self._is_fittable and not self.check_is_fitted():
+        if self.should_fit():
             raise PreprocessorNotFittedException(
                 "`fit` must be called before `transform`."
```

Comment:
should we update this error msg to be:

fit must be called before transform, or simply use fit_transform() to run both steps.